scrapy_poet/injection.py

```diff
@@ -303,7 +304,7 @@ def get_callback(request, spider):
     return request.callback
 
 
-def is_callback_requiring_scrapy_response(callback: Callable):
+def is_callback_requiring_scrapy_response(callback: Callable, request: Request):
```

Comment:
Can we get callback from the request, why is it required to pass both?

Reply:
This is tricky since the original `callback` parameter currently accepts the fact that it could receive a value of **not** `None` since we're providing the spider's `parse()` method if `request.callback=None`.

I've refactored this again to instead accept an optional `raw_callback` value meant to accept the actual value of `request.callback`. I think this is more backward compatible overall. https://github.com/scrapinghub/scrapy-poet/pull/106/commits/e8e22fa700c6c4bcc9d4b878d9ef6cc8c8c7f2b0

The other alternative is to move this as a classmethod to `Injector` so that it has access to the `spider` instance which in turn grants access to the spider's `parse()` method if needed. I think this approach is more limiting than the one above.